src/n64/mod.rs

```diff
@@ -1,13 +1,21 @@
-mod n64;
+mod audio_interface;
 pub mod cpu;
+mod interconnect;
+pub mod mem_map;
+mod n64;
+mod peripheral_interface;
 mod pif;
-mod rsp;
 mod rdp;
-mod audio_interface;
-mod video_interface;
-mod peripheral_interface;
+mod rsp;
 mod serial_interface;
-mod interconnect;
-pub mod mem_map;
+mod video_interface;
 
+pub use self::cpu::Cpu;
+pub use self::interconnect::Interconnect;
 pub use self::n64::N64;
+pub use self::pif::Pif;
+type AudioInterface = self::audio_interface::AudioInterface;
+type PeripheralInterface = self::peripheral_interface::PeripheralInterface;
+type Rsp = self::rsp::Rsp;
+type SerialInterface = self::serial_interface::SerialInterface;
+type VideoInterface = self::video_interface::VideoInterface;
```

Comment:
This stuff seems weird to me though, as a normal "use" didn't work. And I didn't want to make them public either. Almost seems like an oversight in the Rust compiler.